Integrations/python/deephaven/learn/__init__.py

```diff
@@ -0,0 +1,226 @@
+        ValueError : if at least one of the Output columns already exists in the table.
+    """
+
+    inputColumns = [input.input.getColNames()[i] for input in inputs for i in range(len(input.input.getColNames()))]
+
+    if table.hasColumns(inputColumns):
+
+        if not outputs == None:
+            outputColumns = [output.output.getColName() for output in outputs]
+
+            if not table.hasColumns(outputColumns):
+                return
+
+            else:
+                overlap = set(outputColumns).intersection(table.getMeta().getColumn("Name").getDirect())
+                raise ValueError(f"The columns {overlap} already exist in the table. Please choose Output column names that are not already in the table.")
+
+        return
+
+    else:
+        difference = set(inputColumns).difference(table.getMeta().getColumn("Name").getDirect())
+        raise ValueError(f"Cannot find columns {difference} in the table. Please check your spelling and try again.")
+
+
+@_passThrough
+def _createNonconflictingColumnName(column, table):
+    """
+    Ensures that the given column is not present in the table, and modifies it if necessary until it is not in the table.
+
+    Args:
+        column : column name that should not exist in the table.
+        table  : table to check column against.
+
+    Returns:
+        the column name, modified if necessary so that it is not present in the table.
+    """
+
+    if not table.hasColumns(column):
+        return column
+
+    else:
+        i = 0
+        while table.hasColumns(column):
+            column = column + str(i)
+
+            if not table.hasColumns(column):
+                return column
+            i += 1
+
+
+@_passThrough
+def exec(table=None, model_func=None, inputs=[], outputs=[], batch_size = None):
+    """
+    This is the primary tool for linking Deephaven tables with Python deep learning libraries.
+
+    The inputs are used to collect data from the table; then, this data is passed through the model_func for computing,
+    and the results are scattered back into the table with the outputs.
+
+    Here, 
+
+    Args:
+        table      : the Deephaven table to perform computations on.
+        model_func : function that performs computations on the table.
+        inputs     : list of Input objects that determine how data gets extracted from the table.
+        outputs    : list of Output objects that determine how data gets collected back into the table.
+        batch_size : number of rows for which model_func is evaluated at once. Note that this must be provided for a live table.
+
+    Returns:
+        the table with added columns containing the results of evaluating model_func.
+
+    Raises:
+        ValueError : if no batch size was provided.
+        ValueError : if at least one of the Input columns does not exist in the table.
+        ValueError : if at least one of the Output columns already exists in the table.
```

Comment:
Inaccurate if you are also checking for Output columns appearing more than once.